File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-5.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        

        ans = 0
        l = 0
        count = {}
        for r in range(len(s)):
            count[s[r]] = count.get(s[r], 0) + 1
            window = (r - l) + 1
            maxF = max(count.values())
            while window - maxF > k:
                count[s[l]] -= 1
                if count[s[l]] == 0:
                    del count[s[l]]
                l += 1
                window = (r - l) + 1
                maxF = max(count.values())
            ans = max(ans, window)
        return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        int ans = 1;
        unordered_set<int> seen;

        for (int num : nums) {
            if (!seen.count(num)){
                seen.insert(num);
            }
        }

        for (int num : nums) {
            if (!seen.count(num - 1)) {
                int currentNum = num;
                int localCount = 1;
                while (seen.count(currentNum + 1)) {
                    currentNum += 1;
                    localCount += 1;
                }
                ans = max(localCount, ans);
            }
        }
        return ans;
    }
};